File: frontend/src/services/borrowService.js
import api from "./api";

export const getAllBorrowRecords = async () => {
    const response = await api.get("/borrow");
    return response.data;
};

export const getMyBorrowedBooks = async (userId) => {
    const response = await api.get(
        `/borrow/history/${userId}`
    );

    return response.data;
};

export const returnBook = async (borrowId) => {
    const response = await api.put(
        `/borrow/return/${borrowId}`
    );

    return response.data;
};

File: backend/src/main/java/com/LMS/backend/service/BorrowService.java
package com.LMS.backend.service;

import com.LMS.backend.dto.BorrowRequest;
import com.LMS.backend.entity.BorrowRecord;

import java.util.List;

public interface BorrowService {

    BorrowRecord borrowBook(
            BorrowRequest request);

    BorrowRecord returnBook(
            Long borrowId);

    List<BorrowRecord> getAllBorrowRecords();

    List<BorrowRecord> getBorrowHistoryByUser(
            Long userId);

    List<BorrowRecord> getOverdueBooks();

}